### DISCUSSION TITLE
Whatsapp Speech to Text

### DISCUSSION THREAD

**Discussion by altbert** [openai/whisper#996] [Show and tell]
Hi everyone!   
I've made a Whatsapp speech to text transcription using Whisper, Flask and Whatsapp-Web.js   
   
Can be self hosted or using the OpenAI api.  
  
Here it's the code [Whatsapp Speech To Text](https://github.com/altbert/Whatsapp_speech_to_text)    
   
It was a project i was thinking to make, but just recently had the time to make.

**Comment by cijjas** [comment]
hey! Question i am using some snippets of code you have on your repository. I will begin saying im new to this and maybe im doing dumb questions. The thing is, i already know how to convert mp3 files into text with openai's api but i dont understand your SpeechToTextTranscript function works, i find myself unable to understand what is the url that should be called with the request.post. i thought at first its only the whisper api url but its not. im not able to make it work. Can you help me understand what should i put instead of apiHost or the url overall. I would have solved this if i was able to convert base64 wav files into mp3 but im not able too. Thank you for your great work, will be attending to any response!

**Reply by altbert** [reply]
Hi! The apiHost variable is the url of the Flask API that handles the transcription using Whisper. The code for this API can be found in the repo under api/api.py.
The url should be edited to point to the server that you are hosting this API. In the case of this repo the variable is set to point to the right container in the docker-compose.yml.

**Reply by cijjas** [reply]
okay thank you!

**Reply by cijjas** [reply]
And thank you for your work

**Comment by hoehermann** [comment]
Just for fun, I just did the very same [here](https://github.com/hoehermann/whatsmeow-transcribe) with [whatsmeow](https://github.com/tulir/whatsmeow) instead of Whatsapp-Web.js. I am glad we have so many options how to use Whisper!

**Reply by altbert** [reply]
That's really cool!
